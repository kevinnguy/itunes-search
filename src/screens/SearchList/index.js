import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';

import SearchBar from '../../components/SearchBar';
import SongList from '../../components/SongList';

import { fetchSongResults } from '../../utils/networking/api';

export default class SearchList extends Component {
  state = {
    data: null,
    searchQuery: '',
    isLoading: false,
  };

  onChangeText = searchQuery => {
    this.setState({ searchQuery });
  };

  onPressSearch = async () => {
    const { searchQuery } = this.state;
    if (!searchQuery || !searchQuery.length) {
      return;
    }

    this.setState({ isLoading: true });
    const results = await fetchSongResults(searchQuery);
    this.setState({ data: results, isLoading: false });
  };

  onPressRow = songData => {
    this.props.onPressSong(songData);
  };

  render() {
    const { data, isLoading } = this.state;
    return (
      <View style={styles.container}>
        <SearchBar
          onChangeText={this.onChangeText}
          isLoading={isLoading}
          onPress={this.onPressSearch}
        />
        <SongList data={data} onPress={this.onPressRow} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
