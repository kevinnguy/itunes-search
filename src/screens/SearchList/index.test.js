import React from 'react';
import renderer from 'react-test-renderer';

import SearchList from './index';

test('renders correctly', () => {
  const tree = renderer.create(<SearchList />).toJSON();
  expect(tree).toMatchSnapshot();
});
