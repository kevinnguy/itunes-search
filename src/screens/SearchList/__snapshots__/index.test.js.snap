// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#888",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "paddingHorizontal": 20,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      autoFocus={true}
      numberOfLines={1}
      onChangeText={[Function]}
      placeholder="Search artists, songs, albums"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "fontSize": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "purple",
          "borderColor": "purple",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={null}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#888",
              "fontSize": 20,
            }
          }
        >
          No search results
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
